pkg/machinery/imager/quirks/quirks.go

```diff
@@ -222,3 +222,20 @@ func (q Quirks) SupportsUnifiedInstaller() bool {
 
 	return q.v.GTE(minTalosVersionUnifiedInstaller)
 }
+
+// XFSMkfsConfig returns the mkfs.xfs config for the given Talos version.
+func (q Quirks) XFSMkfsConfig() string {
+	switch version := q.v; {
+	// if the version doesn't parse, we assume it's latest Talos
+	case version == nil:
+		return "/usr/share/xfsprogs/mkfs/lts_6.12.conf"
+	case version.GTE(semver.MustParse("1.10.0")):
+		return "/usr/share/xfsprogs/mkfs/lts_6.12.conf"
+	case version.GTE(semver.MustParse("1.9.0")):
+		return "/usr/share/xfsprogs/mkfs/lts_6.6.conf"
```

Comment:
I'm confused a bit on the LTS versions here, 1.9 & 1.10 are both LInux 6.12.x based, and 1.6-1.8 should be 6.6.x ?

Reply:
yeh, but isn't the plan to use 6.6x settings for 1.9.x?

Reply:
probably yes, makes sense, but we'd need to document it here so it's clear what the rules are

Reply:
should we add a quirk to only enable this starting from 1.10, maybe older xfsprogs doesn't have mkfs config files shipped :thinking:

Reply:
> should we add a quirk to only enable this starting from 1.10, maybe older xfsprogs doesn't have mkfs config files shipped 🤔

hold on, this code will run in two contexts:

* in Talos itself, it will run for 1.10+ only
* in the Image Factory, it will run for any version of Talos, and this is where it becomes actually important, and we provide proper defaults.

Reply:
> actually scratch that, downgrades shouldn't re-create STATE/EPHEMERAL so I guess defualts is good

my point they should be able to read STATE/EPHEMERAL, but they should be created on older version though.

Reply:
so my point is the following (roughly) to stay compatbile:

| Talos version | Linux | xfs compat |
|----------|-----|--------|
| 1.10 | 6.12 | 6.12 |
| 1.9 | 6.12 | 6.6 |
| 1.8 | 6.6 | 6.6 |
| 1.7 | 6.6 | 6.1 |
| 1.6 | 6.1 | 6.1 |
| 1.5 | 6.1 | 6.1 |